src/SignalR/clients/ts/signalr/src/HubConnection.ts

```diff
@@ -545,8 +545,18 @@ export class HubConnection {
                     case MessageType.Close:
                         this.logger.log(LogLevel.Information, "Close message received from server.");
 
-                        // We don't want to wait on the stop itself.
-                        this.stopPromise = this.stopInternal(message.error ? new Error("Server returned an error on close: " + message.error) : undefined);
+                        const error = message.error ? new Error("Server returned an error on close: " + message.error) : undefined;
+
+                        if (message.allowReconnect === true) {
+                            // It feels wrong not to await connection.stop() here, but processIncomingData is called as part of an onreceive callback which is not async,
+                            // this is already the behavior for serverTimeout(), and HttpConnection.Stop() should catch and log all possible exceptions.
+
+                            // tslint:disable-next-line:no-floating-promises
+                            this.connection.stop(error);
```

Comment:
How does this interact with someone calling stop on the hub connection?

Reply:
HubConnection.stop() immediately moves the typescript client into the disconnecting state. If HubConnection.stop() were to be called before the HubConnection's connectionClosed handler fired, the connectionClosed handler would see that the HubConnection already transitioned to the disconnecting state and transition to the disconnected state instead of trying to reconnect.

https://github.com/aspnet/AspNetCore/blob/1599a5a2a791def8a23784756c303a914c0e1631/src/SignalR/clients/ts/signalr/src/HubConnection.ts#L660

This already worked because the client needs to be able to reliably handle a connection that just happens to disconnect for any reason (not just close messages) right as someone calls HubConnection.stop()